pipelines/videoPipeline.py

```diff
@@ -0,0 +1,36 @@
+import threading
+from genericPipeline import GenericPipeline
+
+
+class VideoPipeline(GenericPipeline):
```

Comment:
This seems like a lot of copy-pasted code. Perhaps create a parameter `size` in the parent class constructor so that it can be generalized.

Reply:
The issue is that changing the pipeline to use a list in place of one package requires a change in add and get logic. That's why

Reply:
So it could possibly be generalized. A pipeline with a single element can be easily made with a list of size one.

Reply:
Also the locking logic here will only ever allow a single element to be stored anyway.